### PR TITLE
Invite user

### LINKED ISSUE #51: Add user invite functionality
Project owners should be able to invite a user to a project. If the owner has multiple projects he/she will have to select the project which to invite for.

### PR DESCRIPTION
**Pull request by MateyByrd**
Resolves #51 

- Adds the functionality as a project owner to invite a user to your projects.
- You can only invite a user to projects they have not been invited to.
- When you are owner of multiple projects you can select to which project you want to invite the user.

### CODE REVIEW

File: server/routes/api/v1/users.js [hunk 1]
```diff
@@ -256,18 +257,29 @@ router.post('/:id/invitations',
...
+        let joinRequest = new JoinRequests({
+          owner: req.user,
+          project: data[0],
+          invited: data[1]
+        });
+        if (req.body.message) joinRequest.message = req.body.message;
+
+        return Promise.all([
+          joinRequest.save(),
+          Notifications.create({message: `You have been invited to project ${data[0].name}`, owner: req.params.id})
```
Comment (robsel118) [inline server/routes/api/v1/users.js]: Change `owner: req.params.id` to `data[1]` so the user get notified

Comment (MateyByrd) [reply server/routes/api/v1/users.js]: `data[1]` is retrieved by `Users.findById(req.params.id)` so this change would have no effect I think.

Comment (robsel118) [reply server/routes/api/v1/users.js]: Well I don't get the notification with that but with the changes I do

Comment (MateyByrd) [reply server/routes/api/v1/users.js]: Okay, will change it then

**Pull request merged**